src/Build/BackEnd/Components/RequestBuilder/AssemblyLoadsTracker.cs

```diff
@@ -171,7 +171,8 @@ private void CurrentDomainOnAssemblyLoad(object? sender, AssemblyLoadEventArgs a
             if (_loggingContext?.BuildEventContext != null)
             {
                 buildArgs.BuildEventContext = _loggingContext.BuildEventContext;
-                _loggingContext.LogBuildEvent(buildArgs);
+                // Circumvent the logging context validity check since this can happen asynchronously, when the context is being invalidated.
```

Comment:
Can you clarify _why_ the context doesn't have to be valid though? I'm afraid I don't understand.

Reply:
Let's split this from the actuall diag improvement - as then we'll have more info (hopefully) to validate/negate the theory.

So I'll remove this change, merge this, remove fixing indication (as bug will remain) and hpefully very soon we'll have more data to support fix or further investigation.